config/credentials.go

```diff
@@ -0,0 +1,424 @@
+//   - The manager uses fsnotify to watch for file changes and updates the cache automatically
+//   - No pod restart is required - new credentials are used immediately after file update
+//
+// Kubernetes Secret Mount Structure:
+//
+// When a Secret is mounted as a volume, Kubernetes creates this symlink structure:
+//
+//	/secret-mount-path/
+//	├── ..data -> ..2024_01_15_10_30_00.123456   # Symlink to timestamped directory
+//	├── ..2024_01_15_10_30_00.123456/            # Directory with actual secret data
+//	│   ├── token                                # Actual file content
+//	│   └── ca-bundle.crt                        # Actual file content
+//	├── token -> ..data/token                    # Symlink through ..data
+//	└── ca-bundle.crt -> ..data/ca-bundle.crt    # Symlink through ..data
+//
+// During secret rotation, Kubernetes:
+//  1. Creates a new timestamped directory with updated content
+//  2. Atomically swaps the ..data symlink to point to the new directory
+//  3. Deletes the old timestamped directory
+//
+// The individual file symlinks (token, ca-bundle.crt) don't change - only ..data changes.
+// This is why the manager watches for ..data changes to detect secret rotation.
+//
+// Certificate Pool:
+//   - Call InitializeCertPool() to set up system CAs plus custom CA bundles
+//   - Use GetCertPool() to get a clone of the managed certificate pool
+//   - CA bundle files are watched and the pool is rebuilt automatically on changes
+//
+// Usage:
+//
+//	token, err := conf.Credentials.Get(conf.ExternalServices.Prometheus.Auth.Token)
+//	certPool := conf.Credentials.GetCertPool()
+//
+// Internal Data Structures:
+//   - cache: maps absolute file path → credential content (trimmed string)
+//   - watchedDirs: maps directory path → unused struct{} (used as a set of watched directories)
+type CredentialManager struct {
+	mu          sync.RWMutex
+	cache       map[string]string
+	watchedDirs map[string]struct{}
+	watcher     *fsnotify.Watcher
+	done        chan struct{}
+	closeOnce   sync.Once
+
+	// Certificate pool management
+	certPool      *x509.CertPool
+	caBundlePaths []string
+}
+
+// NewCredentialManager creates a new credential manager with file watching enabled.
+// Returns an error if the file watcher cannot be initialized.
+func NewCredentialManager() (*CredentialManager, error) {
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		return nil, fmt.Errorf("failed to create credential file watcher: %w", err)
+	}
+
+	cm := &CredentialManager{
+		cache:       make(map[string]string),
+		watchedDirs: make(map[string]struct{}),
+		watcher:     watcher,
+		done:        make(chan struct{}),
+	}
+
+	go cm.watchFiles()
+	return cm, nil
+}
+
+// Close stops the file watcher and cleans up resources.
+// Should be called during application shutdown.
+func (cm *CredentialManager) Close() {
+	if cm == nil {
+		return
+	}
```

Comment:
Same comment here about `nil` checks. Unless the struct is valid to be used without being initialized, you would expect `var cm *CredentialManager; cm.Close()` to fail with a `nil` pointer exception. You need to call `NewCredentialManager()` to initialize first otherwise something is not being setup properly.